pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -306,3 +307,44 @@ def __init__(self, list_obj) -> None:
         self._order = list_obj.order
         self._class_type = list_obj.class_type
         super().__init__(list_obj)
+
+#==============================================================================
+class ListSort(ListMethod) :
+    """
+    Represents a call to the .sort() method.
+
+    Represents a call to the `.sort()` method, which sorts the elements of the
+    list in ascending order and modifies the original list in place. This means
+    that the elements of the original list are rearranged to be in sorted order.
+    Optional parameters are not supported, therefore they should not be provided. 
+    Note that the .sort() method doesn't return any value.
+    
+    >>> a = [3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5]
+    >>> a.sort()
+    >>> print(a)
+    [1, 1, 2, 3, 3, 4, 5, 5, 5, 6, 9]
+
+    Parameters
+    ----------
+    list_obj : TypedAstNode
+        The list object which the method is called from.
+
+    reverse : TypedAstNode, optional
+        Argument mimicking sort's reverse parameter. This argument is 
+        unsupported so it should not be provided.
+
+    key : FunctionDef, optional
+        A function to specify the sorting criteria(s). This argument is 
+        unsupported so it should not be provided.
```

Comment:
I understand that it is technically complicated to support the `key` argument, but why not supporting `reverse`?

Reply:
There are 2 ways of looking at this:
1. Sort in reverse order
2. Sort, then reverse

In case (1), it is a somewhat similar problem from a technical perspective. @Farouk-Echaref spent some time looking at the STC implementation. I believe he only added what could be supported simply out of the box with STC. We can always increase support later.

In case (2) it should be possible but I don't think that this is the most readable (or the fastest) way to write this.

Reply:
Are we suggesting the user to replace a call to `l.sort(reverse=True)` with these two lines then?
```python
l.sort()
l.reverse()
```
If yes, we should probably write it explicitly in our error message.

Reply:
We did not decide. I was waiting for you to voice an opinion as I don't know if what I wrote:
> I believe he only added what could be supported simply out of the box with STC. We can always increase support later.

is true.

I also don't remember if you have already implemented `reverse`

Reply:
> I believe he only added what could be supported simply out of the box with STC. We can always increase support later.
 
Yes that is the case, as for reverse it's not yet implemented since it doesn't have a direct equivalent in STC or gFTL, we can add support for `sort()` later as you stated when `reverse()` is implemented.

Reply:
In that case I think that the reply to this question:
> Are we suggesting the user to replace a call to l.sort(reverse=True) with these two lines then?

is no. This is simply something that is not supported yet.

Reply:
okey, should I stick with the same error message ?

Reply:
tests for sort updated
af66fca9da2dafca479fad70f0e22565ccee9dbf

Reply:
tests for bool and float added da97dfd9d659967daed2651614dfd7deb2e0c8d5

Reply:
All right.